securesystemslib/schema.py

```diff
@@ -523,7 +523,8 @@ def __init__(self, schema, min_count=0, max_count=sys.maxsize, list_name='list')
   def check_match(self, object):
     if not isinstance(object, (list, tuple)):
       raise securesystemslib.exceptions.FormatError(
-          'Expected ' + repr(self._list_name) + ' but got ' + repr(object))
+          'Expected object of type %s but got type %s'
+          % (type(self._list_name), type(object)))
```

Comment:
According to the docstring `_list_name` is *"A string identifier for the ListOf object."* So I'd...

```suggestion
          % (self._list_name, type(object)))
```
Otherwise you'll always get something like `<class 'str'>`.

Reply:
Maybe it would be nicer to only show the name of the object's class, e.g.`str` instead of `<class 'str'>`...
```suggestion
          % (self._list_name, type(object).__name__))
```
What do you think?

Reply:
I think the new-style  string formatting with `.format()` is generally preferable and we tend to use it in new code. But it's not a blocker.

Reply:
I do prefer the new-style string formatting, but when I couldn't see string formatting in the style guide I decided to err on the side of caution. Especially as I think that `.format()` is only available in Python3  (whilst ssl and its users still want to support Python2)?

Reply:
`format()` was added in 2.6 and we don't support <2.7 so that should be fine. We already use `format()` it in many other places in securesystemslib and in-toto. The code style guideline is a bit outdated in that regard (sorry!). Feel free to update your patch, otherwise I'll merge.

Reply:
Thanks @lukpueh, I've updated to use `format()`